src/vcpkg/binarycaching.cpp

```diff
@@ -914,6 +947,249 @@ namespace
         bool m_interactive;
         bool m_use_nuget_cache;
     };
+    struct GHABinaryProvider : IBinaryProvider
+    {
+        GHABinaryProvider(const VcpkgPaths& paths, bool read, bool write) : paths(paths)
+        {
+            m_read_cache_key = get_environment_variable("VCPKG_GHA_READ_KEY").value_or("vcpkg");
+            m_write_cache_key = get_environment_variable("VCPKG_GHA_WRITE_KEY").value_or("vcpkg");
+
```

Comment:
I have a couple of questions:
1. Is there a reason for the split between read and write?
2. Would it be better to let the key be "vcpkg"?

Reply:
1. GitHub caches can have different keys (the read key can substring match a cache with a different key if that makes sense).
2. Although I can't think of a reason to use different cache keys right now, I don't really see the need to disallow changing them either.

Reply:
Ok in that case I say we don't split them up and let the key be "vcpkg".

Reply:
so leave it as it is now?

Reply:
ummm ok I guess

Reply:
Wait. Why should the user not be able to set the cache key? It may be necessary to deal with different branches and pipelines. Cache capacity is limited at least for free plans.

Reply:
BTW can the cache be updated at all, once created for a given key?

Reply:
> It may be necessary to deal with different branches and pipelines.

Why would you not want a central cache for all branches? Having separate caches would just be unnecessary duplication, at least as far I can tell (maybe I'm missing something though). (I'm not against providing a way for the user to control this though)

> BTW can the cache be updated at all, once created for a given key?

No, caches are immutable. You can manually delete them if you want, or wait until GitHub deletes them after a week.

Reply:
Okay, it is not the final cache key, but just a component.